scripts/variantstore/wdl/extract/run_in_hail_cluster.py

```diff
@@ -0,0 +1,138 @@
+import argparse
+import os
+import uuid
+from google.cloud import dataproc_v1 as dataproc
+from logging import info
+
+
+def configure_logging():
+    import logging
+    import sys
+    # https://stackoverflow.com/a/14058475
+    root = logging.getLogger()
+    root.setLevel(logging.INFO)
+    handler = logging.StreamHandler(sys.stderr)
+    handler.setLevel(logging.INFO)
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    handler.setFormatter(formatter)
+    root.addHandler(handler)
+
+
+def wrap(string):
+    import re
+    return re.sub("\\s{2,}", " ", string).strip()
+
+
+def run_in_cluster(cluster_name, prefix, contig, account, num_workers, worker_machine_type, region, gcs_project,
+                   script_path, vds_url, bed_url, vcf_header_url):
+
+    try:
+        cluster_start_cmd = wrap(f"""
+        
+        hailctl dataproc start 
+         --num-workers {num_workers}
+         --worker-machine-type {worker_machine_type}
+         --region {region}
+         --project {gcs_project}
+         --service-account {account}
+         --num-master-local-ssds 1
+         --num-worker-local-ssds 1 
+         --max-idle=60m
+         --max-age=1440m
+         --subnet=projects/{gcs_project}/regions/{region}/subnetworks/subnetwork
+         {cluster_name}
+         
+        """)
+
+        info(f"Starting cluster '{cluster_name}'...")
+        info(cluster_start_cmd)
+        info(os.popen(cluster_start_cmd).read())
+
+        cluster_client = dataproc.ClusterControllerClient(
+            client_options={"api_endpoint": f"{region}-dataproc.googleapis.com:443"}
+        )
+
+        for cluster in cluster_client.list_clusters(request={"project_id": gcs_project, "region": region}):
+            if cluster.cluster_name == cluster_name:
+                cluster_staging_bucket = cluster.config.temp_bucket
+
+                # THIS IS WHERE YOU CALL YOUR SCRIPT AND COPY THE OUTPUT LOCALLY (to get it back into WDL-space)
+                submit_cmd = wrap(f"""
+                
+                gcloud dataproc jobs submit pyspark {script_path}
+                 --cluster={cluster_name}
+                 --project {gcs_project}
+                 --region={region}
+                 --account {account}
+                 --driver-log-levels root=WARN
+                 --
```

Comment:
What's the dangling '--' about ?

Reply:
I believe that separates arguments to `gcloud dataproc jobs submit` from arguments to the submitted script.